web/components/manalink-card.tsx

```diff
@@ -15,19 +20,21 @@ export type ManalinkInfo = {
 }
 
 export function ManalinkCard(props: {
-  user: User | null | undefined
+  user?: User | null | undefined
   className?: string
   info: ManalinkInfo
-  isClaiming: boolean
+  isClaiming?: boolean
   onClaim?: () => void
+  preview?: boolean
```

Comment:
In general, it's not great to have so many optional props. Makes it less clear for someone using the component how to use it properly. Prefer required props.

One suggestion could be to keep this component private and export two components that wrap it, one for the ManalinkPreview and another for the ManalinkCard. Then you could make all the props required for the exported components. I'm not sure if this is actually the best solution, but think about something like this.

Reply:
Adding button to page instead of components, should take care of this issue